core/templates/dev/head/forms/formBuilder.js

```diff
@@ -895,9 +895,16 @@ oppia.filter('isNonempty', [function() {
 
 oppia.filter('isFloat', [function() {
   return function(input) {
-    // TODO(sll): Accept expressions (like '2.') with nothing after the decimal
-    // point.
-    var FLOAT_REGEXP = /^\-?\d*((\.|\,)\d+)?$/;
+
+    var FLOAT_REGEXP = /^((\-?\d+(\.|\,)?)|(\-?(\.|\,)?))\d+\%?$/;
```

Comment:
Could this be simplified to the following? It looks like the difference between the two alternatives for the prefix are whether there's a digit or not.

     /^\-?\d*(\.|\,)?\d+\%?$/

Also not sure if this accepts things like "2." (should add a test for that). Maybe make the second block of digits optional as well, and add a separate check to ensure that there's at least one digit in the string? (And test stuff like "-", ".", "," to make sure they're invalid.)